sdk/keyvault/keyvault-admin/src/lro/restore/operation.ts

```diff
@@ -0,0 +1,271 @@
+  KeyVaultClientFullRestoreOperationOptionalParams
+} from "../../generated/models";
+import { createSpan, setParentSpan } from "../../tracing";
+import { KeyVaultClientFullRestoreOperationResponse } from "../../generated/models";
+
+/**
+ * An interface representing the publicly available properties of the state of a restore Key Vault's poll operation.
+ */
+export interface RestoreOperationState extends PollOperationState<undefined> {
+  /**
+   * The base URL to the vault
+   */
+  vaultUrl: string;
+  /**
+   * The URI of the blob storage account.
+   */
+  blobStorageUri: string;
+  /**
+   * The SAS token.
+   */
+  sasToken: string;
+  /**
+   * The Folder name of the blob where the previous successful full backup was stored.
+   */
+  folderName: string;
+  /**
+   * Identifier for the full restore operation.
+   */
+  jobId?: string;
+  /**
+   * Status of the restore operation.
+   */
+  status?: string;
+  /**
+   * The status details of restore operation.
+   */
+  statusDetails?: string;
+  /**
+   * The start time of the restore operation in UTC
+   */
+  startTime?: Date;
+  /**
+   * The end time of the restore operation in UTC
+   */
+  endTime?: Date;
+}
+
+/**
+ * An internal interface representing the state of a restore Key Vault's poll operation.
+ * @internal
+ */
+export interface RestorePollOperationState extends PollOperationState<undefined> {
+  /**
+   * Options for the core-http requests.
+   */
+  requestOptions?: RequestOptionsBase;
+  /**
+   * An interface representing the internal KeyVaultClient.
+   */
+  client?: KeyVaultClient;
+  /**
+   * The base URL to the vault
+   */
+  vaultUrl: string;
+  /**
+   * The URI of the blob storage account.
+   */
+  blobStorageUri: string;
+  /**
+   * The SAS token.
+   */
+  sasToken: string;
+  /**
+   * The Folder name of the blob where the previous successful full backup was stored
+   */
+  folderName: string;
+  /**
+   * The id returned as part of the restore request
+   */
+  jobId?: string;
+  /**
+   * Status of the restore operation.
+   */
+  status?: string;
+  /**
+   * The status details of restore operation.
+   */
+  statusDetails?: string;
+  /**
+   * The start time of the restore operation in UTC
+   */
+  startTime?: Date;
+  /**
+   * The end time of the restore operation in UTC
+   */
+  endTime?: Date;
+}
+
+/**
+ * An interface representing a restore Key Vault's poll operation.
+ */
+export interface RestorePollOperation extends PollOperation<RestorePollOperationState, string> {}
+
+/**
+ * Tracing the fullRestore operation
+ */
+async function fullRestore(
+  client: KeyVaultClient,
+  vaultUrl: string,
+  options: KeyVaultClientFullRestoreOperationOptionalParams
+): Promise<KeyVaultClientFullRestoreOperationResponse> {
+  const span = createSpan("generatedClient.fullRestore", options);
+
+  let response: KeyVaultClientFullRestoreOperationResponse;
+  try {
+    response = await client.fullRestoreOperation(vaultUrl, setParentSpan(span, options));
```

Comment:
optional nit: you could do the assignment and return inside the try block if you wanted:

```ts
try {
  const response = await client.fullRestoreOperation(vaultUrl, setParentSpan(span, options));
  return response;
} finally {
   span.end();
}
```

Reply:
Does that still allow the span.end() to run? 😮 I can also just return client.full*

Reply:
returns inside a try still execute the finally, but you'll want to do an `await` or else the time spent inside the promise won't be tracked by the span

Reply:
ohh thank you